examples/ecs/resultful_system_param.rs

```diff
@@ -0,0 +1,103 @@
+//! This example shows how [`ResultfulSystemParam`] can be used to encapsulate simple behavior for use across systems.
```

Comment:
IMO this example should just use the derive macro :)

Reply:
Well, the calculations it does can't be expressed in the derive macro, it just converts types when appropriate. I could change the example entirely to use the the macro, but I felt the example in the docs was sufficient and using the macro again here would have been redundant.

Reply:
Ah alright :)